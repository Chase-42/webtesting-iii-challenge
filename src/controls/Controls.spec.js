// Test away!
import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Controls from "./Controls";
import { fireEvent } from "@testing-library/react";

describe("<Controls /> Component", () => {
  it("Render without crashing", () => {
    render(<Controls />);
  });
  test("<Controls /> snapsnot", () => {
    const wrapper = render(<Controls />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it("Buttons are defined", () => {
    const wrapper = render(<Controls />);
    const lock = wrapper.getByText(/lock/i);
    const open = wrapper.getByText(/close/i);

    expect(lock).toBeDefined();
    expect(open).toBeDefined();
  });
  it("Closed toggle button is disabled if gate is locked", () => {
    const toggleClosed = jest.fn();
    const { getByText } = render(
      <Controls locked={true} toggleClosed={toggleClosed} />
    );
    const closedBtn = getByText(/close gate/i);
    fireEvent.click(closedBtn);
    expect(toggleClosed).not.toHaveBeenCalled();
  });
  it("Locked toggle button is disabled if gate is open", () => {
    const toggleLocked = jest.fn();
    const { getByText } = render(
      <Controls closed={false} toggleLocked={toggleLocked} />
    );
    const lockedBtn = getByText(/lock gate/i);
    fireEvent.click(lockedBtn);
    expect(toggleLocked).not.toHaveBeenCalled();
  });
});

afterEach(cleanup);
