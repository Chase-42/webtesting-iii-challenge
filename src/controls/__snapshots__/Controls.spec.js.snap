// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> <Controls /> snapsnot 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Controls /> Component <Controls /> snapsnot 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;

exports[`match snapshot for controls 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
